subsys/dfu/boot/mcuboot.c

```diff
@@ -57,6 +60,101 @@ struct mcuboot_v1_raw_header {
  * End of strict defines
  */
 
+/**
+ * @brief Get the active slot, i.e. the partition that the current
+ * application is executing from.
+ *
+ * @return currently active slot
+ */
+static int boot_active_slot(void)
+{
+	int slot = 0;
+
+#if FIXED_PARTITION_EXISTS(slot1_partition)
+	if (FIXED_PARTITION_IS_RUNNING_APP_PARTITION(slot1_partition)) {
+		slot = 1;
+	}
+#endif
+#if FIXED_PARTITION_EXISTS(slot2_partition)
+	if (FIXED_PARTITION_IS_RUNNING_APP_PARTITION(slot2_partition)) {
+		slot = 2;
+	}
+#endif
+#if FIXED_PARTITION_EXISTS(slot3_partition)
+	else if (FIXED_PARTITION_IS_RUNNING_APP_PARTITION(slot3_partition)) {
+		slot = 3;
+	}
+#endif
+#if FIXED_PARTITION_EXISTS(slot4_partition)
+	else if (FIXED_PARTITION_IS_RUNNING_APP_PARTITION(slot4_partition)) {
+		slot = 4;
+	}
+#endif
+#if FIXED_PARTITION_EXISTS(slot5_partition)
+	else if (FIXED_PARTITION_IS_RUNNING_APP_PARTITION(slot5_partition)) {
+		slot = 5;
+	}
+#endif
+
+	return slot;
+}
```

Comment:
This is done with minimal code here: subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt.c

Reply:
Yes I originally took the code from there, but this version will support multiple images if/when the support for that is added elsewhere.

Reply:
The code in MCUmgr supports multiple images.

Reply:
It doesn't support multiple images with Direct XIP.

![image](https://github.com/zephyrproject-rtos/zephyr/assets/51752738/ec1bad15-5185-4f82-8240-75096a1bd7f1)

Reply:
> It doesn't support multiple images with Direct XIP.
> 
> ![image](https://user-images.githubusercontent.com/51752738/271518176-ec1bad15-5185-4f82-8240-75096a1bd7f1.png)

Multiple images are not supported with direct XIP.

Reply:
The DIrectXIP has not been tested with multi-image support so far.
Anyway, there is a new function boot_set_next that is supposed to be used for confirming images and it is parametrized in a way that should allow to point slot of pair to confirm.
Cross image confirmation of non-running slot is not advised as it may brick the other core.

Reply:
> The DIrectXIP has not been tested well with multi-image support so far.

I know that probably doesn't work currently, but I think making the behaviour of `boot_write_img_confirmed()` confirm the current slot for single image but the primary slot only for multi-image would make for a confusing API.  If Direct XIP for multi-image doesn't work correctly then the changes here don't make anything worse, they are just allowing for if/when Direct XIP for multi-image is fully supported in the future.

> Anyway, there is a new function boot_set_next that is supposed to be used for confirming images and it is parametrized in a way that should allow to point slot of pair to confirm.

Yep, I've used that new function in this PR.

> Cross image confirmation of non-running slot is not advised as it may brick the other core.

I agree, but the changes here still don't allow that to be done as only `boot_write_img_confirmed()` is changed and that only affects the current image.